Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -1072,6 +1473,608 @@ def get_cim_mapping_field_command():
+    enriched_notables, not_yet_enriched_notables, failed_notables = differentiate_notables(not_yet_enriched_notables)
+    cache_object.not_yet_enriched_notables = not_yet_enriched_notables
+
+    if enriched_notables:
+        if not_yet_enriched_notables:
+            demisto.info('Enriched {} notables successfully. {} left to enrich.'.format(
+                len(enriched_notables), len(not_yet_enriched_notables)))
+        else:
+            demisto.info('Enriched {} notables successfully. No notables left to enrich.'.format(
+                len(enriched_notables)))
+
+    if failed_notables:
+        demisto.info('The following {} notables failed the enrichment process: {}, creating incidents without '
+                     'enrichment.'.format(len(failed_notables), [notable.id for notable in failed_notables]))
+
+
+def differentiate_notables(notables):
+    """ Differentiates the notables list by their status
+
+    Args:
+        notables (list): The list of notables
+
+    Returns:
+        enriched_notables (list): The list of notables that were enriched
+        not_yet_enriched_notables (list): The list of notables that were not yet enriched
+        failed_notables (list): The list of notables that failed to enrich
+
+    """
+    enriched_notables, not_yet_enriched_notables, failed_notables = [], [], []
+
+    for notable in notables:
+        if notable.status == NotableStatus.ENRICHED:
+            enriched_notables.append(notable)
+        elif notable.status == NotableStatus.NOT_YET_ENRICHED:
+            not_yet_enriched_notables.append(notable)
+        elif notable.status == NotableStatus.CREATED_INCIDENT_ENRICH_FAILURE:
+            failed_notables.append(notable)
+
+    return enriched_notables, not_yet_enriched_notables, failed_notables
+
+
+def enrich_fetched_notable(service, notable, num_enrichment_events, incidents, cache_object):
+    """ Enriches fetched notable. Three enrichments possible: Drilldown, Asset & Identity.
+    If all enrichment type executions were unsuccessful, creates a regular incident, Otherwise updates the
+    integration context for the next fetch to handle the open enrichment.
+
+    Args:
+        service (splunklib.client.Service): Splunk service object
+        Possible values are: Drilldown, Asset & Identity.
+        incidents (list): The incident to be submitted at the end of the run.
+        num_enrichment_events (int): The maximal number of events to return per enrichment type.
+        notable (Notable): The notable.
+        cache_object (Cache): The enrichment mechanism cache object
+
+    """
+    notable.data.update({enrichment_type: [] for enrichment_type in ENABLED_ENRICHMENTS})
+    enrichment = Enrichment(notable)
+
+    if DRILLDOWN_ENRICHMENT in ENABLED_ENRICHMENTS:
+        splunk_job = drilldown_enrichment(service, enrichment.notable.data, num_enrichment_events)
+        maintain_enrichment_metadata(enrichment, splunk_job, DRILLDOWN_ENRICHMENT)
```

Comment:
We will add a function to  the Notable class:
```
def add_new_job(self, job: Job):
    self.jobs.append(job)
```
Instead of the `maintain_enrichment_metadata` function will create short function called `create_job_from_splunk_job`:

```
def create_job_from_splunk_job(job_type, splunk_job):
    if not splunk_job:
        status = Job.FAILED
    return Job(job_type=job_type, status=status, job_id=splunk_job["sid"])
```

```suggestion
    if DRILLDOWN_ENRICHMENT in ENABLED_ENRICHMENTS:
        splunk_job = drilldown_enrichment(service, enrichment.notable.data, num_enrichment_events)
        notable.add_new_job(create_job_from_splunk_job(DRILLDOWN_ENRICHMENT, splunk_job)
        
```